mixer/pkg/il/evaluator/evaluator.go

```diff
@@ -91,17 +91,13 @@ func (e *IL) EvalPredicate(expr string, attrs attribute.Bag) (bool, error) {
 }
 
 // EvalType evaluates expr using the attr attribute bag and returns the type of the result.
-func (e *IL) EvalType(expr string, finder expr.AttributeDescriptorFinder) (pb.ValueType, error) {
```

Comment:
Is eval type not in the request path, and therefore no need of caching the evaluated expression ?

Reply:
This version is not in the request path. Type evaluation is done during compilation. It is a separate piece of code.